src/status_im2/navigation/events.cljs

```diff
@@ -27,13 +28,13 @@
   [_ comp-id]
   {:navigate-to-within-stack comp-id})
 
-(rf/defn open-modal
-  {:events [:open-modal]}
-  [{:keys [db]} comp screen-params]
-  {:db            (-> (assoc db :view-id comp)
-                      (all-screens-params comp screen-params))
-   :dispatch      [:hide-bottom-sheet]
-   :open-modal-fx comp})
+(re-frame/reg-event-fx :open-modal
+ (fn [{:keys [db]} [component screen-params]]
+   {:db (-> db
+            (assoc :view-id component)
+            (all-screens-params component screen-params))
+    :fx [[:dispatch [:hide-bottom-sheet]]
+         [:open-modal-fx component]]}))
 
```

Comment:
this looks far cleaner imo. I never understood why we are naming these events as defn's before 👍

Reply:
It does right? @J-Son89, they were named as `defns` (at least) for these reasons I believe:

- The team wanted to standardized defining events with `rf/defn` macro to enforce "global" interceptors before actual global interceptors were introduced in re-frame.
- `rf/merge` wants functions to be called, it needs access to vars, so by default `rf/defn` was created to also help with that, to reduce the boilerplate of having to define functions all the time.

None of those reasons exist anymore when using `:fx`, except we should define functions when we want to call them in unit tests, but this will be an exception, since most events don't need to be unit tested, since they're too simple.